src/chexus/validators.py

```diff
@@ -257,6 +257,33 @@ def validate(self, node: Dataset | Group) -> Violation | None:
         return Violation(node.name)
 
 
+class detector_numbers_unique_in_all_detectors(Validator):
+    def __init__(self) -> None:
+        super().__init__(
+            "detector_numbers are not unique",
+            "The values in all detector_numbers fields in all "
+            "detectors should be unique.",
+        )
+        self._seen_detector_numbers = np.array([], dtype='int')
+
+    def applies_to(self, node: Dataset | Group) -> bool:
+        return (
+            isinstance(node, Group)
+            and node.attrs.get('NX_class') == 'NXdetector'
+            and 'detector_number' in node.children
+        )
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        detector_numbers = node.children['detector_number'].value
```

Comment:
Some files have multi-dim detector numbers. We should flatten before doing the rest of operations, I think?